File: Game_Skeleton/Rational.java
import java.math.BigInteger;

public class Rational {
	long num;
	long denom;
	
	public Rational(long num, long denom)
	{
		this.num = num;
		this.denom = denom;
		reduce();
	}
	
	private void reduce()
	{
		long gcd = gcd(num, denom);
		num /= gcd;
		denom /= gcd;
	}
	
	public Rational add(Rational r)
	{
		return new Rational(num*r.denom + r.num * denom, denom*r.denom);
	}
	
	public Rational multiply(Rational r)
	{
		return new Rational(num*r.num,denom*r.denom);
	}
	
	public boolean equals(Rational r)
	{
		return r.num == num && r.denom == denom;
	}
	
	public boolean equals(long i)
	{
		return num == i && denom == 1;
	}
	
	public long gcd(long a, long b)
	{
		return BigInteger.valueOf(a).gcd(BigInteger.valueOf(b)).longValue();
	}
	
	public boolean lessThan(double d)
	{
		return (num+0.0) / denom < d;
	}
	
	public String toString()
	{
		return toDoubleString();//num+"/"+denom;
	}
	
	public String toDoubleString()
	{
		return (num+0.0)/denom+"";
	}
}

File: v1/Rational.py
class Rational:

	def __init__(self,numerator, denominator):
		self.num = numerator
		self.denom = denominator
		self.reduce()

	def reduce(self):
		gcd = gcd(num, denom)
		self.num //= gcd
		self.denom //= gcd

	def add(self, r):
		return Rational(self.num*r.denom + r.num * self.denom, self.denom*r.denom)

	def multiply(self, r):
		return Rational(self.num*r.num,self.denom*r.denom);

	def equals(self, r):
		return r.num == self.num and r.denom == self.denom;

	def equals(self, i):
		return self.num == i and self.denom == 1

	def gcd(self, x, y):
    	while(y):
           x, y = y, x % y
        return abs(x)

	def lessThan(self, d):
		return (self.num+0.0) / self.denom < d

	def __str__(self):
		return self.toDoubleString()

	def toDoubleString(self):
		return (num+0.0)/denom+"";
